src/main/java/seedu/address/logic/parser/AddTaskParser.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.parser;
+
+import java.time.LocalDateTime;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.IncorrectCommand;
+
+public class AddTaskParser {
+
+    private final AddDeadlineParser addDeadlineParser;
+
+    private final AddEventParser addEventParser;
+
+    private final AddFloatingTaskParser addFloatingTaskParser;
+
+    public AddTaskParser() {
+        addDeadlineParser = new AddDeadlineParser();
+        addEventParser = new AddEventParser();
+        addFloatingTaskParser = new AddFloatingTaskParser();
+    }
+
+    public AddTaskParser(LocalDateTime referenceDateTime) {
+        addDeadlineParser = new AddDeadlineParser(referenceDateTime);
+        addEventParser = new AddEventParser(referenceDateTime);
+        addFloatingTaskParser = new AddFloatingTaskParser();
+    }
+
+    public Command parse(String str) {
+        Command cmd;
+
+        cmd = addEventParser.parse(str);
+        if (!(cmd instanceof IncorrectCommand)) {
+            return cmd;
+        }
+
+        cmd = addDeadlineParser.parse(str);
+        if (!(cmd instanceof IncorrectCommand)) {
+            return cmd;
+        }
+
+        return addFloatingTaskParser.parse(str);
```

Comment:
Revised by adding a new `incorrectCommand` in `AddTaskParser`. (in next PR)
But the solution is still redundant, will think up a better solution in the future.